### PR TITLE
"dimmed" nodes/links on hover

### LINKED ISSUE #15: Reduce Opacity of Ignored Nodes on Hover
https://github.com/jollyjerr/mindgraph/assets/51095001/d28f83ad-0eab-4ce8-bbed-036066bb28a1

### PR DESCRIPTION
**Pull request by mpryor**
This feature is one take on how issue https://github.com/jollyjerr/mindgraph/issues/15 could be handled.

It really just adds a `isAnythingActive` (very open to a different name) parameter to the drawable interface that a user could then handle however they want.

This introduces a default configuration to the Mindgraph elements that utilize this new parameter as well.

this pr closes #15

### CODE REVIEW

File: packages/draw/src/renderables.ts [hunk 1]
```diff
@@ -26,13 +26,15 @@ export class RenderableLink implements Drawable {
     );
   }
 
-  draw(canvas: Canvas, isActive: boolean): void {
+  draw(canvas: Canvas, isActive: boolean, isAnythingActive: boolean): void {
```
Comment (jollyjerr) [inline packages/draw/src/renderables.ts:29]: This is probs dumb but what if we do `isActive?: boolean` with:

```
true = bright
undefined = normal
false = dimmed
```

Comment (jollyjerr) [reply packages/draw/src/renderables.ts:29]: and remove isAnythingActive

Comment (mpryor) [reply packages/draw/src/renderables.ts:29]: Would that change the logic at https://github.com/jollyjerr/mindgraph/pull/17/files#diff-85da206e66d53748eee13584274555243888de57e6a7d1e3e45e4a4f08e13cc7R126 to be something like:
```
if (activeDrawables.includes(d) || activeDrawables.length > 0) {
  d.draw(this, activeDrawables.includes(d))
} else {
  d.draw(this)
}
```
?

Comment (mpryor) [reply packages/draw/src/renderables.ts:29]: That feels like there's some hidden logic that the end-user would have to be aware of. 

But I see where you're coming from, we're only really trying to encode 1 of 3 states in a single value. 

Maybe the second parameter to `draw` is just a `highlight` type with 3 possible values "dimmed" | "active" | "normal", or just an `enum``?

```
enum Highlight {
  Active
  Dimmed
  Normal
}
```

Comment (jollyjerr) [reply packages/draw/src/renderables.ts:29]: Let's do that! I normally prefer string literals to enums (just simpler), but either way works

```ts
draw(canvas: Canvas, highlight: "active" | "dimmed" | "normal")
```

Comment (mpryor) [reply packages/draw/src/renderables.ts:29]: Cool! I don't think I have a preference for one over the other so I'll try that out

**Review by jollyjerr — APPROVED** [review]
This is a great starting spot! Three things i'd like to see eventually (definitely don't have to be part of this pr)

1. Deal with the overlap #18 
2. Animate the transition so it's not as jarring
3. Also dim the text of dimmed nodes


https://github.com/jollyjerr/mindgraph/assets/51095001/b40b4a31-ccc3-4b3b-a29a-d81d15ad4079

**Comment by jollyjerr** [comment]
Oh also could you run a changeset on this so it's documented 🤗

**Comment by mpryor** [comment]
> Animate the transition so it's not as jarring

Agree! I was trying to figure out why it felt so abrupt, but that's totally it.

**Comment by jollyjerr** [comment]
I think we could use https://github.com/d3/d3-transition to get some transitions going on

**Pull request merged**